packages/flutter/lib/src/foundation/stack_frame.dart

```diff
@@ -190,6 +190,11 @@ class StackFrame {
       return stackOverFlowElision;
     }
 
+    assert(
+      line != '===== asynchronous gap ===========================',
+      'Got a stack frame from package:stack_trace, where a vm or web frame was expected.'
```

Comment:
Maybe we could add an explanation here about how to use the new static global to fix this.